src/ORM/DataList.php

```diff
@@ -308,72 +313,106 @@ public function distinct($value)
     }
 
     /**
-     * Return a new DataList instance as a copy of this data list with the sort
-     * order set.
+     * Return a new DataList instance as a copy of this data list with the sort order set
      *
-     * @see SS_List::sort()
-     * @see SQLSelect::orderby
+     * Raw SQL is not accepted, only actual field names can be passed
+     *
+     * @param string|array $args
      * @example $list = $list->sort('Name'); // default ASC sorting
-     * @example $list = $list->sort('Name DESC'); // DESC sorting
+     * @example $list = $list->sort('Name ASC, Age DESC');
      * @example $list = $list->sort('Name', 'ASC');
-     * @example $list = $list->sort(array('Name'=>'ASC', 'Age'=>'DESC'));
-     *
-     * @param string|array Escaped SQL statement. If passed as array, all keys and values are assumed to be escaped.
-     * @return static
+     * @example $list = $list->sort(['Name' => 'ASC', 'Age' => 'DESC']);
      */
-    public function sort()
+    public function sort(...$args): static
     {
-        $count = func_num_args();
-
+        $count = count($args);
         if ($count == 0) {
```

Comment:
Updated

I've added in support for passing in `null` to clear any existing sorts on DataList so that it's consistent with [limit(null)](https://github.com/silverstripe/silverstripe-framework/blob/4.11/src/ORM/Queries/SQLSelect.php#L277)

It'll throw an exception if you `sort([])`, `sort('')`, `sort(false)` or `sort(0)` or other scalars